godot-core/src/builtin/array.rs

```diff
@@ -581,59 +782,59 @@ impl GodotFfi for Array {
     }
 }
 
-#[repr(C)]
-pub struct TypedArray<T> {
-    opaque: OpaqueArray,
-    _phantom: PhantomData<T>,
-}
-impl<T> TypedArray<T> {
-    fn from_opaque(opaque: OpaqueArray) -> Self {
-        Self {
-            opaque,
-            _phantom: PhantomData,
-        }
-    }
-}
-
-impl<T> Clone for TypedArray<T> {
-    fn clone(&self) -> Self {
-        unsafe {
-            Self::from_sys_init(|self_ptr| {
-                let ctor = ::godot_ffi::builtin_fn!(array_construct_copy);
-                let args = [self.sys_const()];
-                ctor(self_ptr, args.as_ptr());
-            })
+/// Allows for construction of [`TypedArray`] literals, much in the same way as Rust's standard
+/// `vec!` macro. The type of the array is inferred from the arguments.
+///
+/// # Example
+///
+/// ```
+/// let arr = array![3, 1, 4];
+/// ```
+///
+/// To create an `Array` of variants, you need to convert each element explicitly:
+///
+/// ```
+/// let arr: Array = array![3.to_variant(), 1.to_variant(), 4.to_variant()];
+/// ```
```

Comment:
Do these pass doctests?
Because the Godot engine needs to be running, which it isn't during `cargo test` 🤔 

If this causes problems, you can mark the sections as `no_run`.

Reply:
No, and done :)